### ISSUE TITLE
Setting loading-timeout to 0 doesn't wait for children to load or generate a warning.

### ISSUE THREAD

**Issue by ppeyro** [PredixDev/px-context-browser#59]
## Expected behavior and actual behavior:

API doc: set loading-timeout to 0  to wait forever to load:
1.   That's not the case, there is no wait to load children if loading-timeout is set to 0. As soon as the open arrow is selected, it returns you back to the parent.
2.  If the behavior is implemented to actually "wait forever", we should add a warning alerting the developer.  

## Environment :
CB: Redesign branch, today's pull
App-helpers: Master branch, today's pull

Browsers: Chrome

## Screenshot

![screen shot 2017-07-11 at 12 58 51 pm](https://user-images.githubusercontent.com/10835611/28087876-d7fc6ec6-6638-11e7-9703-b90d93adc142.png)

## Code example
```
        <px-context-browser
            items="{{data}}"
            selected="{{selectedItem}}"
            has-filter
            open-arrow
            has-favorites
            loading-timeout=0
            enable-warnings
            >
        </px-context-browser>
```

/cc @davidrleonard

**Comment by davidrleonard** [comment]
@ppeyro This should be fixed in 793b9c5c264fcbf05a77526ea73a2bc64afce083.

The new behavior works like this:

1. If `loading-timeout` is not specified by the developer, it defaults to `5000` milliseconds. The context browser waits 5 seconds for children to be provided — if none are provided, the loading is canceled. The developer can cancel loading before the timeout by calling the `cancelLoading()` method on the context browser element.
2. If `loading-timeout` is specified by the developer and it is a number greater than `0`, the context browser does the same thing as in 1, but waits as many milliseconds as the developer requested.
3. If `loading-timeout` is the number `0`, the context browser will wait forever for children. The developer can still call `cancelLoading()` to cancel the loading event.
4. If `loading-timeout` is not a valid non-negative number the context browser will throw an error to the console and fail.

**Comment by davidrleonard** [comment]
Additional fix (to throw the error) as of 7775033bdfaf1594165621e2b61a788ab82ef051
